src/helpers/validation.ts

```diff
@@ -25,3 +28,24 @@ export const followersSchema = yup.object({
   follower: yup.string().required("followerId can't be blank"),
   following: yup.string().required("ownerId can't be blank"),
 });
+
+export const userIdSchema = yup.object({
+  userId: yup.string().required(),
+});
+
+export const tweetIdSchema = yup.object({
+  tweetId: yup.string().required(),
+});
+
+export const likesQuerySchema = yup.lazy((values) => {
+  if (values.userId) {
+    return userIdSchema;
+  } else if (values.tweetId) {
+    return tweetIdSchema;
+  } else {
+    return yup.object({
+      userId: yup.string().required("tweetId or userId are required"),
+      tweetId: yup.string().required("tweetId or userId are required"),
+    });
+  }
+});
```

Comment:
You don't need to make it conditional validation, just put all query params into single schema, and don't make them required